src/components/Review/Review.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react'
+import {Rate,Modal,Button,Input,Alert} from "antd";
+import './Review.css';
+import {connect} from 'react-redux';
+import { compose } from 'redux';
+import {postReviewAction} from '../../redux/actions/reviewActions';
+const {TextArea}  = Input
+
+class Review extends Component {
+    state = {
+        loading: false,
+        visible: false,
+        rating: null,
+        review: null
+      };
+    ratingChange = (e)=>{
+        console.log(e);
+        this.setState({
+            rating:e
+        })
+    }
+    changeReview = (e)=>{
+        e.preventDefault()
+        this.setState({
+            review:e.target.value
+        })
+    }
+      showModal = () => {
+        this.setState({
+          visible: true,
+        });
+      };
+    
+      handleOk = () => {
+        const {_id} = this.props.service;
+        const {rating,review}  = this.state
+        console.log(_id,review,rating);
+        this.props.postReviewAction(_id,{rating,review});
+
+
+      };
+    
+      handleCancel = () => {
+        this.setState({ visible: false
+         });
+      };
+    
+    render() {
+        const { visible, loading,rating } = this.state;
+        // console.log(rating);
+        const {error,review}  = this.props;
+        // console.log(review);
+        console.log(error,review);
+        return (
+          <div id="review">
+            
+            <Button type="primary" onClick={this.showModal}  style={{margin:20}}>
+              Review the Service
+            </Button>
+            <Modal
+              visible={visible}
+              title="Review the service"
+              onOk={this.handleOk}
+              onCancel={this.handleCancel}
+              footer={[
+                <Button key="back" onClick={this.handleCancel}>
+                  Return
+                </Button>,
+                <Button key="submit" type="primary" loading={loading} onClick={this.handleOk} disabled={rating===null?true:false}>
+                  Submit
+                </Button>,
+              ]}
+            >
+              {rating===null ?
+            (<Alert type="info" message="Please atleast rate the service before submitting" />):null}
```

Comment:
Not required. Just keep the submit button disabled.

Reply:
done